lighthouse-core/gather/gather-runner.js

```diff
@@ -258,7 +236,8 @@ class GatherRunner {
       // Abuse the passContext to pass through gatherer options
       passContext.options = gathererDefn.options || {};
       const artifactPromise = Promise.resolve().then(_ => gatherer.pass(passContext));
-      gathererResults[gatherer.name].push(artifactPromise);
+      const gathererResult = gathererResults[gatherer.name] || [];
```

Comment:
can we ts-ignore if this is not a real thing/really fix if it is? adding the promise to an array that doesn't go anywhere seems worse haha :)

Reply:
> adding the promise to an array that doesn't go anywhere seems worse haha :)

ha...that is much worse. I'll fix

Reply:
ok, slightly more verbose, but ts-ignoring would unfortunately handle not only the defined case but also wrong types for `push()`, and these are two out of three places we push to this array, so that seemed like it was defeating the purpose of better types :)

Unfortunately the compiler won't recognize the one-line check `if (!gathererResults[gatherer.name]) throw new Error(...)` as properly guarding the undefined case.

Reply:
ah well we tried! fix you got now wfm :)